### PR TITLE
fix: note pan min/max can be up to 128

### PR DESCRIPTION
**Pull request by zacanger**
Note pan can be -128—128. Tested simply by adding one note on the piano roll and using the piano roll pan editor all the way up/down. On FL v 20.9.2 (build 2459), Producer/All Plugins edition, on Apple Silicon mac 12. Test FLP attached.
[test-flp.zip](https://github.com/demberto/PyFLP/files/9124952/test-flp.zip)

### CODE REVIEW

**Comment by demberto** [comment]
Thanks for your contributions :)

I am actually doing an entire rewrite rn, none of those commits are pushed but, I am going for a more standard property based approach. `_FLObject` was long pending a refactor anyways.

Does FL Studio on Mac use UTF16 as well?

**Comment by tochibedford** [comment]
Is the rewrite going to cause issues with already written contributions @demberto

**Comment by zacanger** [comment]
Good catch, I never really use Notebook2 or project info, so didn't notice that. Tested [this commit](c44b87bb99ae558f85a425ea1142b336118d4b38) with a Notebook2.

**Comment by demberto** [comment]
@zacanger `TextEvent`s aren't encoded in UTF-8 in Mac, I just verified it with your test FLP. Notebook2's `DataEvent` might be using it on Mac. Also, the note pan min/max limits seem to be raised in 20.9 because my test FLP has similar notes with max/min pan.

> Please let me know if that change is idiomatic
They will be idiomatic as well, since now I am better at Python myself. But most importantly, the property values will not be generated by all the `parseprop` and `parse_event` mess. They will be deserialized by the property getters from corresponding event data and serialized back by the setters. I have created helper prototypes to reduce code duplication created by this.

Along with this, after the events are all collected into a list by `Parser`, they won't be sent to the `_FLObject`s one by one as they are now. I will introduce event collectors in between which will gather events required for creating every model. After the collectors create their own sublists, I will pass them to constructors of the model classes. These constructors won't do any operations on them and store them in a private variable to be used by the getter/setters.

Or, I have been thinking of other way where the model classes (Channel, Track etc.) could be just simple dataclasses and serialization/deserialization would fall in the hands of collectors.

Also, I need to be very careful about validators in the future, probably discard their use completely. FLP is a very broken format at its core. The fact that they get invoked while parsing (its a bug) is the source of many problems.

**Comment by zacanger** [comment]
Thanks. I reverted the TextEvent change, left the one in Notebook2, but didn't touch DataEvent itself. I definitely look forward to your changes. I'm not surprised the FLP format is messy, since they've kept it mostly backwards-compatible for the last 24 years. It must be quite a chore keeping up with whatever Image Line are doing without good documentation on the format, so really, huge thanks for doing the work in this project :)

File: pyflp/pattern/note.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ class PatternNote(_FLObject):
     128 for MIDI dragged into the piano roll.
     """
 
-    pan: int = _IntProperty(min_=-64, max_=64)
+    pan: int = _IntProperty(min_=-128, max_=128)
```
Comment (demberto) [inline pyflp/pattern/note.py:65]: Max should be 127

**Review by demberto — CHANGES_REQUESTED** [review]

File: pyflp/pattern/note.py (hunk 1 above)
Comment (zacanger) [reply pyflp/pattern/note.py:65]: Whoops, right you are!

**Pull request merged**